titus-server-master/src/main/java/com/netflix/titus/master/appscale/service/DefaultAppScaleManager.java

```diff
@@ -144,7 +146,14 @@ public DefaultAppScaleManager(AppScalePolicyStore appScalePolicyStore, CloudAlar
         this.scalableTargets = new ConcurrentHashMap<>();
         this.metrics = new AppScaleManagerMetrics(registry);
         this.appScaleActionsSubject = PublishSubject.<AppScaleAction>create().toSerialized();
-        this.appScaleActionsSub = appScaleActionsSubject.observeOn(awsInteractionScheduler).subscribe(new AppScaleActionHandler());
+
+        this.appScaleActionsSub = appScaleActionsSubject
+                .onBackpressureDrop(appScaleAction -> {
+                    logger.error("Dropping {}",appScaleAction);
+                    metrics.reportDroppedRequest();
+                })
+                .observeOn(awsInteractionScheduler, ASYNC_HANDLER_BUFFER_CAPACITY)
+                .subscribe(new AppScaleActionHandler(), e -> logger.error("Exception in appScaleActionsSubject ", e));
```

Comment:
I think we need to re-subscribe on errors here, otherwise the rx pipeline just terminates.